### PR TITLE
SLCORE-478 Handle reviewing for security hotspots

### PR DESCRIPTION
**Pull request by damien-urruty-sonarsource**

### CODE REVIEW

File: client-api/src/main/java/org/sonarsource/sonarlint/core/clientapi/backend/hotspot/ChangeHotspotStatusParams.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonarsource.sonarlint.core.clientapi.backend.hotspot;
+
+public class ChangeHotspotStatusParams {
+  private final String configurationScopeId;
+  private final String hostpotKey;
```
[the remark this answers is not in the document]
Comment (damien-urruty-sonarsource) [reply client-api/src/main/java/org/sonarsource/sonarlint/core/clientapi/backend/hotspot/ChangeHotspotStatusParams.java]: Fixed! Thank you

File: client-api/src/main/java/org/sonarsource/sonarlint/core/clientapi/backend/hotspot/HotspotService.java [hunk 2]
```diff
@@ -30,4 +30,44 @@ public interface HotspotService {
 
   @JsonRequest
   CompletableFuture<CheckLocalDetectionSupportedResponse> checkLocalDetectionSupported(CheckLocalDetectionSupportedParams params);
+
+  /**
+   * The list of available statuses differs between SonarQube and SonarCloud, so different values will be returned based on the connectionId:
+   * <ul>
+   *   <li>For SonarCloud, the allowed statuses are {@link HotspotStatus#TO_REVIEW}, {@link HotspotStatus#SAFE} and {@link HotspotStatus#FIXED}</li>
+   *   <li>For SonarQube, on top of the previous ones, the {@link HotspotStatus#ACKNOWLEDGED} is also allowed</li>
```
Comment (nquinquenel) [inline client-api/src/main/java/org/sonarsource/sonarlint/core/clientapi/backend/hotspot/HotspotService.java:38]:
```suggestion
   *   <li>For SonarQube, on top of the previous ones, the status {@link HotspotStatus#ACKNOWLEDGED} is also allowed</li>
```

File: server-api/src/main/java/org/sonarsource/sonarlint/core/serverapi/hotspot/HotspotApi.java [hunk 3]
```diff
@@ -63,6 +67,23 @@ public static boolean permitsTracking(boolean isSonarCloud, Supplier<Version> se
...
   }
 
+  public CompletableFuture<Void> changeStatusAsync(String hotspotKey, HotspotReviewStatus status) {
+    if (helper.isSonarCloud()) {
+      // hotspots are not supported on SonarCloud at the moment
+      return CompletableFuture.completedFuture(null);
+    }
+    var isReviewed = status.isReviewed();
+    var webApiStatus = isReviewed ? "REVIEWED" : "TO_REVIEW";
+    var body = "hotspot=" + urlEncode(hotspotKey) + "&status=" + webApiStatus;
```
Comment (nquinquenel) [inline server-api/src/main/java/org/sonarsource/sonarlint/core/serverapi/hotspot/HotspotApi.java:77]: I'd suggest encoding every parameter of the request

**Review by nquinquenel — APPROVED** [review]
LGTM!

**Pull request merged**